snovault/elasticsearch/indexer_utils.py

```diff
@@ -251,14 +262,29 @@ def locate_link_to(schema_cursor):
 
             # Collect diffs from all possible item_types
             all_possible_diffs = diffs.get(base_field_item_type, [])
+
+            # A linkTo target could be a child type (in that we need to look at parent type diffs as well)
+            # NOTE: this situation doesn't actually occur in our system as of right now
+            # but theoretically could
             parent_types = child_to_parent_type.get(base_field_item_type, None)
             if parent_types is not None:
-                all_possible_diffs += [diffs.get(parent) for parent in parent_types]
+                for parent_type in parent_types:
+                    parent_diff = diffs.get(parent_type, None)
+                    if parent_diff:
+                        all_possible_diffs.extend(parent_diff)
```

Comment:
Again untested, and assuming you don't need the `parent_types` variable later:
```suggestion
           for parent_type in child_to_parent_type.get(base_field_item_type, []):
                all_possible_diffs.extend(diffs.get(parent_type, []))
```